components/brave_wallet_ui/components/desktop/views/explore_web3/web3_dapp_details.tsx

```diff
@@ -0,0 +1,202 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import { Redirect, useParams } from 'react-router'
+
+import ProgressRing from '@brave/leo/react/progressRing'
+
+// Hooks
+import {
+  useGetMainnetsQuery,
+  useGetTopDappsQuery
+} from '../../../../common/slices/api.slice'
+
+// Utils
+import Amount from '../../../../utils/amount'
+import { CurrencySymbols } from '../../../../utils/currency-symbols'
+import { getLocale } from '../../../../../common/locale'
+import { WalletRoutes } from '../../../../constants/types'
+import { openTab } from '../../../../utils/routes-utils'
+
+// Components
+import { NavButton } from '../../../extension/buttons'
+import { CreateNetworkIcon } from '../../../shared/create-network-icon'
+
+// Styles
+import { Column, Row, Text } from '../../../shared/style'
+import {
+  DappCategoryLabel,
+  DappMetricContainer
+} from './web3_dapp_details.styles'
+
+export const DappDetails = () => {
+  // routing
+  const { dappId } = useParams<{
+    dappId: string
+  }>()
+
+  // queries
+  const { isLoading, data: dapp } = useGetTopDappsQuery(undefined, {
+    selectFromResult: (res) => ({
+      isLoading: res.isLoading,
+      data: dappId
+        ? res.data?.find(({ id }) => id === parseInt(dappId))
+        : undefined
+    })
+  })
+
+  const { data: networks } = useGetMainnetsQuery()
+
+  // render
+  if (isLoading) {
+    return (
+      <Column
+        fullHeight
+        fullWidth
+      >
+        <ProgressRing />
+      </Column>
+    )
+  }
+
+  // Dapp not found, redirect to the list
+  if (!dapp) {
+    return <Redirect to={WalletRoutes.Web3} />
+  }
+
+  return (
+    <Column
+      fullHeight
+      fullWidth
+      justifyContent='flex-start'
+    >
+      <Column
+        gap={'8px'}
+        fullWidth
+        justifyContent='flex-start'
+        alignItems='center'
+        margin={'32px 0px 0px 0px'}
+      >
+        <img
+          src={`chrome://image?${dapp.logo}`}
```

Comment:
can we validate `dapp.logo` scheme is `https`? i see http is disallowed in https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ui/webui/sanitized_image_source.cc;l=177?q=kchromeuiimageurl&ss=chromium%2Fchromium%2Fsrc but not sure what happens if it's some other scheme like javascript:.